### PR TITLE
Brezovec conversion I

### PR DESCRIPTION
**Pull request by h-mayorquin**
First draft that covers the conversion of Fitrac data.  The fictract data comes in a binary file encoded in utf8. To read lazily I used polars https://www.pola.rs/ to test how to conver to dynamic tables when the files do not fit in memory (in this case it does).

There is a header that the page of fictract mentions that is not available:
https://github.com/rjdmoore/fictrac/tree/master

We will have to ask the authors about it.

Also, still thinking about how to organize this. Right now, all the data for fictrac is in the same processing module separated with different time series. Same sampling rate. Maybe we could use compass for some of them.

I will add notes for the conversion and some general issue soon.

### CODE REVIEW

File: src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py [hunk 1]
```diff
@@ -1,22 +1,245 @@
...
+from neuroconv.utils.json_schema import FolderPathType, FilePathType
+from neuroconv.tools.nwb_helpers import get_module
+from pynwb.behavior import SpatialSeries
+import polars as pl
+
 
 from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import DeepDict
 
 class BrezovecBehaviorInterface(BaseDataInterface):
```
Comment (bendichter) [inline src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py:14]: Why not call this FictrackInterface?

File: src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py [hunk 2]
```diff
@@ -1,22 +1,245 @@
...
         # All the custom code to write to PyNWB
 
+        descriptions = BrezovecBehaviorInterface.column_to_description
+        columns = list(descriptions)[:23]
+
+        int_type_columns = BrezovecBehaviorInterface.int_type_columns
+        column_to_type = lambda x: pl.Int64 if x in int_type_columns else pl.Float64
+
+        polars_schema = {column: column_to_type(column) for column in columns}
+        df_fitrac = pl.scan_csv(
```
Comment (bendichter) [inline src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py:79]: How big are these CSV files? Usually they can be read in memory with no problem, but I support it's possible to have a file big enough where lazy reading would be necessary. Would it be possible to add an implementation option that does not require polars?

File: src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py (hunk 1 above)
Comment (h-mayorquin) [reply src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py:14]: Yes, that makes sense. There is nothing specific here so far. I will do it.

Comment (CodyCBakerPhD) [reply src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py:14]: 'FicTracInterface`, no K and capital T

from spelling on main package: https://github.com/rjdmoore/fictrac

Comment (h-mayorquin) [reply src/clandinin_lab_to_nwb/brezovec/brezovecbehaviorinterface.py]: Done.

File: src/clandinin_lab_to_nwb/brezovec/brezovec_convert_session.py [hunk 3]
```diff
@@ -23,29 +22,16 @@ def session_to_nwb(data_dir_path: Union[str, Path], output_dir_path: Union[str,
...
-    source_data.update(dict(Sorting=dict()))
-    conversion_options.update(dict(Sorting=dict()))
-
-    # Add Behavior
-    source_data.update(dict(Behavior=dict()))
-    conversion_options.update(dict(Behavior=dict()))
+    # Add Fictrac
+    file_path = data_dir_path / "fictrac" / "fictrac-20200228_161226.dat"
+    source_data.update(dict(FicTrac=dict(file_path=str(file_path))))
+    conversion_options.update(dict(FicTrac=dict()))
```
Comment (bendichter) [inline src/clandinin_lab_to_nwb/brezovec/brezovec_convert_session.py:28]: Do you need to add an empty dict here?

File: src/clandinin_lab_to_nwb/brezovec/brezovec_metadata.yaml [hunk 4]
```diff
@@ -9,8 +9,9 @@ NWBFile:
...
     - Druckmann, Saul
-    - Clandinin, Thomas 
+    - Clandinin, Thomas
 Subject:
-  species: Drosophila
-  age: TBD  # in ISO 8601, such as "P1W2D"
-  sex: TBD  # One of M, F, U, or O
+  subject_id: TBD  # A unique identifier for the subject
+  species: Drosophila melanogaster
+  age: 3-4D  # in ISO 8601, such as "P1W2D"
```
Comment (bendichter) [inline src/clandinin_lab_to_nwb/brezovec/brezovec_metadata.yaml:16]: What are you trying to say here? Between 3 and 4 days old? that would be "3D/4D"

File: src/clandinin_lab_to_nwb/brezovec/brezovec_convert_session.py (hunk 3 above)
Comment (h-mayorquin) [reply src/clandinin_lab_to_nwb/brezovec/brezovec_convert_session.py:28]: No, inherited from the cookiecutter. Removing.

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Will anything on this PR change with the merge of FicTrac interface on NeuroConv? Or should we just go ahead with this one here and update later?

**Pull request merged**